docs/cookbook/links.rst

```diff
@@ -0,0 +1,239 @@
+.. _ref_cookbook_links:
+
+=====
+Links
+=====
+
+Link items define a specific relationship between two object types.
+Links have a direction, but can be traversed in both ways :ref:`forward and
+backward <ref_cookbook_links_fw_bw>`.
+
+
+Movie Example
+=============
+
+To define a relationship we use ``link`` keyword:
+
+.. code-block:: edgeql
+
+    module default {
+        type Movie {
+            required property title -> str;
+            required link director -> Person;
+            multi link cast -> Person;
+        }
+        type Person {
+            required property first_name -> str;
+            required property last_name -> str;
+        }
+    }
+
+In terms of SQL database we defined two foreign keys on the table containing
+movies. But link is much more than that.
+
+
+Nested Sets
+-----------
+
+Links allow fetching relatioships using single query:
+
+.. code-block:: edgeql-repl
+
+    tutorial> select Movie {
+    .........    director: { first_name },
+    .........    cast: { first_name },
+    ......... };
+    {
+        Object {
+            director: Object { first_name: 'Denis' },
+            cast: {
+                Object { first_name: 'Harrison' },
+                Object { first_name: 'Ryan' },
+                Object { first_name: 'Ana' },
+            }
+        }
+    }
+
+Aggregates
+----------
+
+Similarly you can run some aggregates on the nested sets:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Movie { title, cast_size:=count(.cast) };
+    {Object { title: 'Blade Runner 2049', cast_size: 3 }}
+
+You can find more information on aggregates in the
+:ref:`Cookbook <ref_cookbook_aggregates>` and the reference of
+:ref:`set functions <ref_eql_operators_set>` (or just search for "aggregate").
+
+
+.. _ref_cookbook_links_bw:
+
+Backward Links
+--------------
+
+In the movie example above, we have only shown a forward link traversal:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Movie { title, cast: { first_name } };
+    {
+        Object {
+            title: 'Blade Runner 2049',
+            cast: {
+                Object { first_name: 'Harrison' },
+                Object { first_name: 'Ryan' },
+                Object { first_name: 'Ana' },
+            }
+        }
+    }
+
+Here is a another example of using forward link. This time we only return
+last names of the artists as plain string (not an object). In this case, we
+need to alias the field with ``:=``. This example uses **forward link** ``.>``
+operator to make the code clearer:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Movie {
+    .........     title,
+    .........     starring := Movie.>cast.last_name,
+    ......... };
+    {Object {
+        title: 'Blade Runner 2049',
+        starring: {
+            'Ford',
+            'Gosling',
+            'de Armas',
+        }
+    }}
+
+To find all movies that a person is starred in we use a **backward link**
```

Comment:
>  we use a **backward link**

I'm more in favor of using "forward link traversal" and "reverse link traversal".  Adding "traversal" is important, because technically there is no such thing as a "backward link" -- there is only one link, you just traverse it in both directions.